src/CanvasRenderingContext2d.cc

```diff
@@ -2643,3 +2639,26 @@ NAN_METHOD(Context2d::Ellipse) {
   }
   cairo_set_matrix(ctx, &save_matrix);
 }
+
+bool checkArgs(const Nan::FunctionCallbackInfo<Value> &info, double *args, int argsNum, int offset){
+  int argsEnd = offset + argsNum;
+  bool areArgsValid = true;
+
+  for (int i = offset; i < argsEnd; i++) {
+    double val = info[i]->NumberValue();
+
+    if (areArgsValid) {
+      if (isnan(val) || isinf(val)) {
```

Comment:
(continued from previous comment) ...

GCC and clang are marginally better with `isinf`, but MSVC is significiantly better with `val == std::numeric_limits<double>::infinity() || val == -std::numeric_limits<double>::infinity()`. MSVC doesn't inline `isinf` in this case, and has a different implementation:

```
(inline)
00007FFBE4F48DF5  ucomisd     xmm6,xmm7  
00007FFBE4F48DF9  jp          checkArgs+8Dh (07FFBE4F48DFDh)  
00007FFBE4F48DFB  je          checkArgs+93h (07FFBE4F48E03h)  

vs

(call)
00007FFBE4F5C854  mov         rcx,7FFFFFFFFFFFFFFFh  
00007FFBE4F5C85E  movsd       mmword ptr [rsp+8],xmm0  
00007FFBE4F5C864  mov         rax,qword ptr [_bitsy]  
00007FFBE4F5C869  and         rax,rcx  
00007FFBE4F5C86C  mov         rcx,7FF0000000000000h  
00007FFBE4F5C876  sub         rax,rcx  
00007FFBE4F5C879  shr         rax,3Fh  
00007FFBE4F5C87D  ret  
```

GCC and Clang: https://godbolt.org/g/GxxiqG


```
MSVC benchmarks:
  (ops/sec)      master     orig PR  ==infinity
lineTo()     22,447,439  20,242,779  21,345,058
arc()         1,883,895   1,747,626   1,872,457
strokeRect() 13,206,246  12,629,641  13,189,635
```

I guess leave it as-is since more servers run Linux and Windows.

/ramble